### PR TITLE
mininet: 2.3.0d6 -> 2.3.0

### PR DESCRIPTION
**Pull request by teto**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabaff — COMMENTED** [review]
`fullName = ...` for the license should be [updated](https://github.com/mininet/mininet/blob/master/LICENSE#L1) too.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126470` at 9d57cca9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>mininet</li>
    <li>python38Packages.mininet-python</li>
    <li>python39Packages.mininet-python</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

help2man is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/virtualization/mininet/default.nix:27:3:
```
   |
27 |   buildInputs = [ python which help2man ];
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/virtualization/mininet/default.nix:27:3:
```
   |
27 |   buildInputs = [ python which help2man ];
   |   ^
```
</li>
  </ul>
</details>

**Comment by teto** [comment]
I've moved help2man to nativeBuildInputs but which is actually necessary at runtime and it's probably true for other packages so that check may need to be removed.

File: pkgs/tools/virtualization/mininet/default.nix [hunk 1]
```diff
@@ -9,22 +9,24 @@ let
...
   };
 
   buildFlags = [ "mnexec" ];
   makeFlags = [ "PREFIX=$(out)" ];
 
   pythonPath = [ python.pkgs.setuptools ];
-  buildInputs = [ python which help2man ];
+  nativeBuildInputs = [ help2man ];
+
+  propagatedBuildInputs = [ python which ];
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/mininet/default.nix:29]: Python3?

File: pkgs/tools/virtualization/mininet/default.nix [hunk 2]
```diff
@@ -40,7 +42,7 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Emulator for rapid prototyping of Software Defined Networks";
     license = {
-      fullName = "Mininet 2.3.0d6 License";
+      fullName = "Mininet 2.3.0 License";
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/mininet/default.nix:45]: Is the licensee free?

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -30084,7 +30084,7 @@ in
 
   scotch = callPackage ../applications/science/math/scotch { };
 
-  mininet = callPackage ../tools/virtualization/mininet { };
+  mininet = callPackage ../tools/virtualization/mininet { python = python3; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:30087]:
```suggestion
  mininet = callPackage ../tools/virtualization/mininet { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**